CAIPs/caip-n.md

```diff
@@ -0,0 +1,84 @@
+
+While some new providers are being written without protocol prefixes, it is trivial to add a prefix to all methods of any given JSON-RPC interface, making it easy for a new provider to be constructed from a provider that is compatible with this proposal, and making it trivial for this proposal to wrap any given JSON-RPC interface.
+
+## Motivation
+
+The Ethereum JSON-RPC protocol has established [JSON-RPC](https://www.jsonrpc.org/specification) as the standard for new blockchains, with new chains introducing their own JSON-RPC interfaces:
+- [Cosmjs/json-rpc](https://www.npmjs.com/package/@cosmjs/json-rpc)
+- [Starkware](https://www.npmjs.com/package/starkware-provider)
+- [Filecoin](https://docs.filecoin.io/reference/lotus-api/)
+
+For wallets looking to support multiple protocols at once, a popular approach for the short-term has been for wallets to manage multiple networks (like in [EIP-2015](https://github.com/ChainAgnostic/CAIPs.git)), but this has drawbacks:
+- It puts a UX burden on users to think about networks.
+- It prevents the creation of multi-chain applications, since each is connected to only a single network.
+- It creates race conditions in applications where the provider's chain could change definition while a method call is in-flight.
+
+Two other approaches would be:
+- Allow the provider to return additional providers in response to methods.
+- Allow the provider to handle multiple simultaneous networks at once itself.
+
+Allowing a provider to return additional providers is a nice developer experience, but it breaks outisde the scope of JSON-RPC, and so I would argue it should be built at the convenience library layer, and instead the JSON-RPC interface itself should define how to handle multiple providers.
+
+## Specification
+
+The specification is very simple:
+
+Providers that want to support multiple networks should use unique prefixes to namespace the methods for those various networks. As those prefixes and underlying interfaces are defined, they should be publicized so that they can be more widely adopted by other providers and wallets.
+
+Additionally, the provider should allow detecting what interfaces are supported, and so I will suggest one feature detection method, which will return an array of the supported interface identifiers:
+
+```
+{
+  method: 'getSupportedProtocols',
+}
+
+// returns:
+['EtherMain', 'EtherRopsten', 'EtherXDai', 'Bitcoin']
+```
+
+## Rationale
+
+Opening this CAIP as a discussion point, there are many variations that could be adopted to the proposal; Maybe this proposal itself should be behind a namespace, for example. Otherwise, I think the Motivation above covers the rationale adequatel.
```

Comment:
```suggestion
Opening this CAIP as a discussion point, there are many variations that could be adopted to the proposal; Maybe this proposal itself should be behind a namespace, for example. Otherwise, I think the Motivation above covers the rationale adequately.
```